### ISSUE TITLE
rez env with an escaped > symbol in a package requirement still redirects stdout on Windows

### ISSUE THREAD

**Issue by michaelmovies** [nerdvegas/rez#342]
Cannot test this on Linux, but on Windows it's certainly an issue.

Because Rez requirements can include the < and > stream redirection characters, obviously there's a problem if these are not escaped in the shell when passed to Rez:

```
rez env python>2.7.11
```

The above will run "rez env python" and pipe all output to a new file "2.7.11" ... the actual version requirement is lost. So of course the user must run the escaped version, which on Windows uses the ^ char:

```
rez env python^>2.7.11
```
This actually works and the Rez configured environment does pick up ie 2.7.12 (or fails if no greater version than 2.7.11 is installed) ... however it still creates the "2.7.11" file.

That tends to indicate that it's either happening in a Python subprocess call (where the unescaped arg is being passed and shell=True), or it's happening in the compiled shell code.  Having a hell of a time finding where or which it is.

**Comment by michaelmovies** [comment]
Hah! Just found it.  The compiled shell code includes a section that sets the REZ_USED_REQUEST env var to the exact string value received ... that then runs "set REZ_USED_REQUEST=python>2.7.11", and as that's run as an individual command it redirects the STDOUT of the set command to that file, creating the new file in the process.

That also explains why an escaped < char results in a "The system cannot find the file specified" message when the file being redirected to stdin doesn't exist.

The other implication is that something similar could happen with the "implicit_packages" variable in rez_config.py, as you can set soft requirements with the > and < chars there as well.

Well, at least I know where it is.  I'll see if I can find time to dig in and fix.

M

**Comment by nerdvegas** [comment]
Mm, is this actually a case of Rex not escaping the string?

You can test this with the rez-interpret tool:

# in test.rex
env.FOO = 'bah>5.4'

]$ rez-interpret test.rex --format bash
export FOO="bah>5.4"

Presumably the same codepath for Windows is not escaping that output
correctly.

A

**Comment by michaelmovies** [comment]
Yeah, I think I found it, the escape_string method currently does nothing. Working on a merge request.

M

**Comment by michaelmovies** [comment]
This is now addressed in #341.
